src/librustdoc/core.rs

```diff
@@ -90,11 +86,22 @@ impl<'tcx> DocContext<'tcx> {
         ret
     }
 
-    crate fn enter_resolver<F, R>(&self, f: F) -> R
-    where
-        F: FnOnce(&mut resolve::Resolver<'_>) -> R,
-    {
-        self.resolver.borrow_mut().access(f)
+    crate fn resolve_from_cache(
+        &self,
+        path: &str,
+        ns: Namespace,
+        module_id: DefId,
+    ) -> Result<collect_intra_doc_links::Res, ()> {
+        self.resolver_caches
+            .doc_link_resolutions
+            .get(&(path.to_string(), ns, module_id))
+            .copied()
+            // .unwrap_or_else(|| panic!("no resolution for {:?} {:?} {:?}", path, ns, module_id))
+            .unwrap_or_else(|| {
+                eprintln!("no resolution for {:?} {:?} {:?}", path, ns, module_id);
+                Err(())
+            })
```

Comment:
```suggestion
            .unwrap_or_else(|| {
                warn!("no resolution for {:?} {:?} {:?}", path, ns, module_id);
                Err(())
            })
```
Are you sure the warning is necessary at all? Rustdoc should handle invalid links, I don't think this is really an error in the sense that it should be reported unconditionally.